src/com/xilinx/rapidwright/interchange/DeviceResourcesWriter.java

```diff
@@ -241,8 +241,8 @@ public static void writeAllSiteTypesToBuilder(Device device, DeviceResources.Dev
                 SitePin.Builder pin = pins.get(j);
                 pin.setName(allStrings.getIndex(pinNames.get(j)));
                 pin.setDir(j <= highestIndexInputPin ? Direction.INPUT : Direction.OUTPUT);
-                Integer siteWireIdx = site.getSiteWireIndex(pinNames.get(j));
-                pin.setSitewire(siteWireIdx == null ? -1 : siteWireIdx);
+                String tileWireName = site.getTileWireNameFromPinName(pinNames.get(j));
```

Comment:
You are correct in that this is not writing out the site wire name associated with the site pin.  However, the `Site.getTileWireNameFromPinName()` method will return the tile wire from the context of the site which does not match the schema in this situation.  Because Vivado (and thus RapidWright) do not have SiteType objects, we are using Site objects.  But we are writing out a SiteType object.  We want the site wire name, which could be gotten by:

```
                Integer siteWireIdx = site.getSiteWireIndex(pinNames.get(j));
                String siteWireName = siteWireIdx == null ? null : site.getSiteWireName(siteWireIdx);
                pin.setSitewire(siteWireName == null ? -1 : allStrings.getIndex(siteWireName));
```

Reply:
Ya, fixed by removing Sitewire, as it can be reached `SiteWire`.